### PR TITLE
Use CSS files for custom properties and media

### PR DESCRIPTION
**Pull request by mor10**
Replaces `cssVariables.json` with two new files:

- `assets/css/custom-properties.css` for custom properties
- `assets/css/custom-media.css` for custom media queries

## Description

Addresses issue #278.

With this change, developers write proper CSS custom properties and custom queries and PostCSS processes them dynamically. This means developers can choose to disable the rewrite PostCSS does and instead just enqueue the new files directly to get custom properties and media working in the browser.

Because of some sort of magic, `require-uncached` is not necessary to get dynamic updates in the `npm run dev` process, which is great because `require-uncached` was coughing up a lot of errors.

## List of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:
- [x] This pull request relates to a ticket.
- [x] This code is tested.
- [x] This change has been added to CHANGELOG.md
- [x] I want my code added to WP Rig.

### CODE REVIEW

File: gulp/constants.js [hunk 1]
```diff
@@ -66,6 +65,8 @@ let paths = {
 		dest: `${rootPath}/`
 	},
 	styles: {
+		cssCustomProperties: `${rootPath}/assets/css/src/custom-properties.css`,
```
Comment (ataylorme) [inline gulp/constants.js]: Can you use `assetsDir` instead of `rootPath`?

**Comment by ataylorme** [comment]
@mor10 this looks great! One minor nitpick is all.

**Comment by bamadesigner** [comment]
@mor10 I lean towards them being in the same file. There's no need for an extra file for that. They should be together.

**Comment by ataylorme** [comment]
> I lean towards them being in the same file. There's no need for an extra file for that. They should be together.

I respectfully disagree but my personal preference is to modularize things. Using WP Rig for a project I would add more variables and media queries, making one monolithic file get quite large.

**Comment by ataylorme** [comment]
Also previously with JSON we could pick variables and media queries from the same file and pass them separately into the corresponding parser. I'm not sure we can do that from a single CSS file.

**Comment by mor10** [comment]
@ataylorme Just tested: A single file works fine.

**Pull request merged**